mailpoet/assets/js/src/newsletter_editor/blocks/coupon/usage_limits.tsx

```diff
@@ -0,0 +1,80 @@
+import { Panel, PanelRow, PanelBody, TextControl } from '@wordpress/components';
+import Backbone from 'backbone';
+import { Component } from '@wordpress/element';
+import { MailPoet } from 'mailpoet';
+
+type Props = {
+  getValueCallback: (name: string) => string | boolean | Backbone.Collection;
+  setValueCallback: (
+    name: string,
+    value: string | boolean | Backbone.Collection,
+  ) => void;
+};
+
+type State = {
+  usageLimit: string;
+  usageLimitPerUser: string;
+};
+
+class UsageLimits extends Component<Props, State> {
+  private readonly getValueCallback: (
+    name: string,
+  ) => string | boolean | Backbone.Collection;
+
+  private readonly setValueCallback: (
+    name: string,
+    value: string | boolean | Backbone.Collection,
+  ) => void;
+
+  constructor(props: Props) {
+    super(props);
+    this.getValueCallback = props.getValueCallback;
+    this.setValueCallback = props.setValueCallback;
+
+    this.state = {
+      usageLimit: this.getValueCallback('usageLimit') as string,
+      usageLimitPerUser: this.getValueCallback('usageLimitPerUser') as string,
+    };
+  }
+
+  render() {
+    return (
+      <Panel>
+        <PanelBody
+          title={MailPoet.I18n.t('usageLimits')}
+          className="mailpoet-coupon-block-usage-limits"
+          initialOpen={false}
+        >
+          <PanelRow>
+            <TextControl
+              className="mailpoet_field_coupon_usage_limits_per_coupon"
```

Comment:
I added those classes automatically because I didn't know if I needed them in one of my following improvements, such as design or validation. I can remove them.